services/scabbard/libscabbard/src/service/factory.rs

```diff
@@ -491,15 +719,35 @@ mod tests {
     }
 
     fn get_factory() -> ScabbardFactory {
-        ScabbardFactory::new(
-            Some("/tmp".into()),
-            Some(1024 * 1024),
-            Some("/tmp".into()),
-            Some(1024 * 1024),
-            #[cfg(feature = "diesel-receipt-store")]
-            ":memory:".to_string(),
-            Arc::new(Mutex::new(Box::new(Secp256k1Context::new()))),
-        )
+        #[cfg(not(all(feature = "diesel-receipt-store", feature = "sqlite")))]
+        let receipt_store_factory_config = ScabbardFactoryStorageConfig::Lmdb {
+            db_dir: "/tmp".into(),
```

Comment:
Instead of using /tmp here, we should use TempDir to create a directory for us - https://docs.rs/tempdir/0.3.7/tempdir/

Reply:
I will add a commit, as this was using the same config as the original test.